### PR TITLE
[CL Message Audit]: MsgCreateFullRangePositionAndSuperfluidDelegate

### PR DESCRIPTION
**Pull request by czarcas7ic**
Clickup Checklist:
https://app.clickup.com/37420681/v/dc/13nzm9-19430/13nzm9-35650

## What is the purpose of the change

This PR includes changes made from the internal audit of MsgCreateFullRangePositionAndSuperfluidDelegate.

## Brief Changelog

- `mintSharesLockAndUpdate` is now required to be a full range position (before we just assumed it would only be called for full range positions)
- Fixed gas fee now only charged on tick initialization (before was being charged on tick updates as well)
- Expands existing tests
- Adds tests for previously untested methods:
  - TicksToSqrtPrice
  - UpdateLiquidityIfActivePosition
  - GetConcentratedLockupDenomFromPoolId
- Added "Create Full Range Position and Superfluid Delegate" message to superfluid readme 
- Adds respective CLI command
  - Changes validateBasic logic that would have prevented setting a CL denom as a superfluid asset

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/lp_test.go [hunk 1]
```diff
@@ -1071,12 +1070,12 @@ func (s *KeeperTestSuite) TestSendCoinsBetweenPoolAndUser() {
 		"only asset0 is greater than sender has, position creation (user to pool)": {
 			coin0:       sdk.NewCoin("eth", sdk.NewInt(100000000000000)),
 			coin1:       sdk.NewCoin("usdc", sdk.NewInt(1000000)),
-			expectError: true,
+			expectedErr: fmt.Errorf("insufficient funds"),
```
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:1073]: nit: we can have a test-global definition to be reused

File: x/concentrated-liquidity/math/tick_test.go [hunk 2]
```diff
@@ -197,6 +197,101 @@ func (suite *ConcentratedMathTestSuite) TestTickToSqrtPrice() {
 	}
 }
 
+func (suite *ConcentratedMathTestSuite) TestTicksToSqrtPrice() {
```
Comment (p0mvn) [inline x/concentrated-liquidity/math/tick_test.go:200]: Hmmm I thought I've already reviewed this before??

File: x/concentrated-liquidity/position_test.go [hunk 3]
```diff
@@ -1192,6 +1212,239 @@ func (s *KeeperTestSuite) TestMintSharesLockAndUpdate() {
 	}
 }
 
+func (s *KeeperTestSuite) TestPositionHasActiveUnderlyingLock() {
```
Comment (p0mvn) [inline x/concentrated-liquidity/position_test.go:1215]: Is this overlapping with another PR? Skipping since I've already reviewed

**Review by p0mvn — APPROVED** [review]
Nice work, super clean

File: x/concentrated-liquidity/position_test.go (hunk 3 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/position_test.go:1215]: Yes, that was my bad. Sorry about that

File: x/concentrated-liquidity/lp_test.go (hunk 1 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/lp_test.go]: Added https://github.com/osmosis-labs/osmosis/pull/5082/commits/a59c8af08ac0d831a2221f4073962fc4da2e92fa

**Pull request merged**